scripts/postprocessing/calc_lattice_strain.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@author: rjk5987 , dcp5303
+
+This script calculates 'lattice strains', elastic strains along a specified sample direction
+averaged over subsets of elements (grains) belonging to different crystallographic fibers.
+
+Designed to take in 'raw_LatticeOrientation' outputs from adios2_extraction.py
+and 'SampleElasticStrain' outputs from strain_Xtal_to_Sample.py
+
+As-is, also requires that 'ElementVolume' was output from adios2_extraction.py
+for use in a volume-weighted average. Typically does not make a significant difference,
+so the user may wih to remove this below.
+
+Saves lattice strains in ASCII text format (.txt) for use in other scripts.
+Output file named 'lattice_strains' and arranged as step # by lattice plane.
+
+Recommended to configure script in an interactive shell, then run from the command line.
+
+User inputs marked with #!!!
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+# Need a materials.h5 file from HEXRD!!!
+from hexrd import rotations as rot
+from hexrd import material , valunits
+
+# makes console print-outs nicer, can CTRL+F and remove these to eliminate this dependency
+from sty import bg , ef , fg
+
+def load_pdata(filename , d_min , energy , mat): 
+
+    kev  = valunits.valWUnit('beam_energy' , 'energy' , energy , 'keV')
+    dmin = valunits.valWUnit('dmin' , 'length' , d_min , 'angstrom')
+    mats = material.load_materials_hdf5(filename , dmin = dmin , kev = kev)
+    pd   = mats[mat].planeData
+
+    return pd
+
+# conversions between radians and degrees
+d2r = np.pi / 180.
+r2d = 180. / np.pi
+
+#%% Booleans.
+
+# whether to plot the resulting lattice strains vs. cycle when finished
+plot = True #!!!
+# whether to save plot (will save to the same directory as the lattice strain file)
+save_plot = False #!!!
+
+#%% Establish filepaths and directories.
+
+# requires a materials.h5 file created in HEXRD
+mat_file = '/dir/to/materials.h5' #!!!
+# material of interest in material file
+mat = 'IN625' #!!!
```

Comment:
@rjk5987 potentially within a comment in this file could you show briefly how to create one of those materials files using the HEXRD api or if there is an example in the HEXRD repo point to it here? It would just make it easier for those unfamiliar with HEXRD to be able to get the bare minimum required to run these scripts aka people like me.